pydbgen/pydbgen.py

```diff
@@ -45,266 +69,151 @@ def license_plate(self,seed=None,style=None):
+
+        domain = choice(self.domain_list)
+
+        name_formats = ["{f}{last}", "{first}{last}",
+                        "{first}.{l}", "{first}_{l}",
+                        "{first}.{last}", "{first}_{last}",
+                        "{last}_{first}", "{last}.{first}"]
+        name_fmt_choice = choice(name_formats)
+        name_combo = name_fmt_choice.format(f=f_name[0], l=l_name[0], first=f_name, last=l_name)
+
+        if choice_int < 7:
+            email = name_combo + '@' + str(domain)
         else:
-            random_int = randint(11,99)
-            result+=name_combo+str(random_int)+'@'+str(domain)
+            random_int = randint(11, 99)
+            email = name_combo + str(random_int) + '@' + str(domain)
 
-        return result
+        return email
 
-    def city_real(self,seed=None):
+    def city_real(self, seed=None):
         '''
         Picks and returns a random entry out of 385 US cities
         seed: Currently not used. Uses seed from the pydb class if chosen by user
         '''
-        import os
-        from six import moves
-        import ssl
-        import random
-        from random import randint,choice
         random.seed(self.seed)
 
-        path = "US_Cities.txt"
-        if not os.path.isfile(path):
-            context = ssl._create_unverified_context()
-            moves.urllib.request.urlretrieve("https://raw.githubusercontent.com/tflearn/tflearn.github.io/master/resources/US_Cities.txt", path)
-        
-        city_list = []
-        fh = open(path)
-        for line in fh.readlines():
-            city_list.append(str(line).strip())    
-        
-        fh.close() 
+        return choice(self.city_list)
 
-        return (choice(city_list))        
-
-    def gen_data_series(self,num=10,data_type='name',seed=None):
+    def gen_data_series(self, num=10, data_type='name', seed=None):
         """
         Returns a pandas series object with the desired number of entries and data type
-        
-        Data types available: 
+
+        Data types available:
         - Name, country, city, real (US) cities, US state, zipcode, latitude, longitude
         - Month, weekday, year, time, date
-        - Personal email, official email, SSN 
+        - Personal email, official email, SSN
         - Company, Job title, phone number, license plate
-        
-        Phone number can be two types: 
+
+        Phone number can be two types:
         'phone_number_simple' generates 10 digit US number in xxx-xxx-xxxx format
         'phone_number_full' may generate an international number with different format
 
         seed: Currently not used. Uses seed from the pydb class if chosen by user
 
         """
-        if type(data_type)!=str:
-            print("Data type not understood. No series generated")
-            return None
+        if type(data_type) != str:
+            raise ValueError("Data type must be of type str, found " + str(type(data_type)))

```

Comment:
Printing an error and returning `None` can be unintuitive in non-interactive scenarios. Raising an exception will make argument validation very explicit.